core/audio/AudioEngine.cpp

```diff
@@ -216,6 +216,28 @@ void AudioEngine::setVolume(AUDIO_ID audioID, float volume)
     }
 }
 
+void AudioEngine::setPitch(AUDIO_ID audioID, float pitch)
+{
+    auto it = _audioIDInfoMap.find(audioID);
+    if (it != _audioIDInfoMap.end())
+    {
+        if (pitch < 0.5f)
```

Comment:
better code: `pitch = std::clamp(pitch, 0.5f, 2.0f);`

Reply:
I was following the pattern of [`setVolume`](https://github.com/axmolengine/axmol/blob/53811d6df6ac768763e8e88126085f2329e9b3ae/core/audio/AudioEngine.cpp#L202) method. Let me know if you still want to go with `std::clamp`.

Reply:
There is no need to follow the patterns of existing code, and any new code should use the best practice at the time it is written.